gprofiler/metadata/application_identifiers.py

```diff
@@ -304,13 +335,18 @@ def get_app_id(self, process: Process) -> Optional[str]:
     ],
 }
 
+_SPARK_IDENTIFIERS_MAP = _IDENTIFIERS_MAP.copy()
+_SPARK_IDENTIFIERS_MAP["java"].insert(0, _JavaSparkApplicationIdentifier())
```

Comment:
I think it's cleaner if you add `java_spark` to the mapping:
```
_IDENTIFIERS_MAP["java_spark"] = _IDENTIFIERS_MAP["java"] + [_JavaSparkApplicationIdentifier()]
```

and then you just pass `java_spark` from `get_java_app_id` if Spark is requested

Reply:
Much cleaner